pennylane/gradients/pulse_gradient.py

```diff
@@ -90,19 +90,26 @@ def _split_evol_ops(op, ob, tau):
         before_t = jax.numpy.array([t0, tau])
         after_t = jax.numpy.array([tau, t1])
 
-    eigvals = qml.eigvals(ob)
-    coeffs, shifts = zip(*generate_shift_rule(eigvals_to_frequencies(tuple(eigvals))))
+    if qml.pauli.is_pauli_word(ob):
+        prefactor = next(iter(qml.pauli.pauli_sentence(ob).values()))
```

Comment:
lol nice!

Reply:
Could consider adding functionality to obtain this directly somehow in the pauli module 😅

Reply:
Yeah... I'll ask around whether this is needed elsewhere.